doc/source/whatsnew/v0.20.0.txt

```diff
@@ -122,6 +122,16 @@ Other enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. _whatsnew.api_breaking.io_compat
+
+Pickle and HDF5 compat requires >= 0.13.0
```

Comment:
Yes, that's certainly better! 
My comment here was mainly about the title (which people scan, so it would be nice if this is clear). The too long version would be something like "Reading pickle and HDF5 files requires them being saved with pandas >= 0.13" or "Possible incompatibility with pickle and HDF5 files created with pandas < 0.13.0", but thinking on more condensed version but that is still clear